godot-core/src/init/mod.rs

```diff
@@ -119,6 +119,73 @@ unsafe extern "C" fn ffi_deinitialize_layer<E: ExtensionLibrary>(
     });
 }
 
+#[cfg(since_api = "4.3")]
+unsafe fn register() {
+    use crate::registry::plugin::PluginItem;
+    use std::collections::{hash_map::Entry, HashMap};
+    #[derive(Default)]
+    struct DocPieces {
+        /// base, description, members
+        definition: [&'static str; 3],
+        /// methods, signals, constants
+        inherent: [&'static str; 3],
+        imethods: &'static str,
+    }
+
+    let mut map = HashMap::<&'static str, DocPieces>::new();
+    crate::private::iterate_plugins(|x| match x.item {
+        PluginItem::InherentImpl { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().inherent = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                inherent: docs,
+                ..Default::default()
+            })),
+        },
+        PluginItem::ITraitImpl { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().imethods = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                imethods: docs,
+                ..Default::default()
+            })),
+        },
+        PluginItem::Struct { docs, .. } => match map.entry(x.class_name.as_str()) {
+            Entry::Occupied(x) => x.into_mut().definition = docs,
+            Entry::Vacant(x) => drop(x.insert(DocPieces {
+                definition: docs,
+                ..Default::default()
+            })),
+        },
+    });
+    for (
+        class,
+        DocPieces {
+            definition: [base, desc, members],
+            inherent: [methods, signals, constants],
+            imethods,
+        },
+    ) in map
+    {
+        let brief = desc.lines().next().unwrap_or_default();
+        let xml = format!(
+            r#"
+<?xml version="1.0" encoding="UTF-8"?>
+<class name="{class}" inherits="{base}" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:noNamespaceSchemaLocation="../class.xsd">
+<brief_description>{brief}</brief_description>
+<description>{desc}</description>
+<methods>{methods}{imethods}</methods>
+<constants>{constants}</constants>
+<signals>{signals}</signals>
+<members>{members}</members>
+</class>"#
+        );
+        // SAFETY: the godot binding is initialized
+        crate::sys::interface_fn!(editor_help_load_xml_from_utf8_chars_and_len)(
+            xml.as_ptr().cast(),
+            xml.len() as _,
+        );
```

Comment:
There's no unsafe block here so it's a bit confusing that you have a safety comment here. If you want to document all the unsafe sections of the function properly then you should add `#[deny(unsafe_op_in_unsafe_fn)]` to this function. As otherwise it's easy to miss. 

We havent fully landed on a policy of when to use `unsafe_op_in_unsafe_fn`, but imo this function seems like it might be a good case. But it's a bit up to you.